garden/src/features/entrypoints/components/modals/PaperModal.tsx

```diff
@@ -46,11 +52,106 @@ const PaperModal = ({ edit, onSave, initialData, trigger }: PaperModalProps) =>
         })) || [],
       doi: initialData?.doi || "",
       citation: initialData?.citation || "",
+      url: initialData?.url || "",
     },
   });
 
+  const url = form.watch("url");
+
+  // Reset form when modal is opened or closed
+  useEffect(() => {
+    if (isOpen) {
+      // If we're editing, load the initial data
+      if (edit && initialData) {
+        form.reset({
+          title: initialData.title || "",
+          authors:
+            initialData.authors?.map((author) => ({
+              value: author,
+              label: author,
+            })) || [],
+          doi: initialData.doi || "",
+          citation: initialData.citation || "",
+          url: initialData.url || "",
+        });
+      } else {
+        // If we're adding a new paper, reset to empty values
+        form.reset({
+          title: "",
+          authors: [],
+          doi: "",
+          citation: "",
+          url: "",
+        });
+      }
+      previousUrlRef.current = "";
+    }
+  }, [isOpen, edit, initialData, form]);
+
+  useEffect(() => {
+    const fetchArxivData = async (arxivId: string) => {
+      setIsLoadingMetadata(true);
+      try {
+        const metadata = await fetchArxivMetadata(arxivId);
+        
+        let fieldsUpdated = false;
+        
+        // Only auto-fill empty fields
+        if (metadata.title && !form.getValues("title")) {
+          form.setValue("title", metadata.title);
+          fieldsUpdated = true;
+        }
+        
+        if (metadata.doi && !form.getValues("doi")) {
+          form.setValue("doi", metadata.doi);
+          fieldsUpdated = true;
+        }
+        
+        if (metadata.citation && !form.getValues("citation")) {
+          form.setValue("citation", metadata.citation);
+          fieldsUpdated = true;
+        }
+        
+        // Check if authors field is empty
+        const currentAuthors = form.getValues("authors");
+        const isAuthorsEmpty = !currentAuthors || currentAuthors.length === 0;
+        
+        if (metadata.authors && metadata.authors.length > 0 && isAuthorsEmpty) {
+          form.setValue(
+            "authors",
+            metadata.authors.map((author) => ({
+              value: author,
+              label: author,
+            }))
+          );
+          fieldsUpdated = true;
+        }
+        
+        if (fieldsUpdated) {
+          toast.success("Paper metadata auto-filled from arXiv");
+        } else {
+          toast.info("No new metadata to fill from arXiv");
```

Comment:
Does the user need a toast in this case?

Reply:
nope!